Basic-Car-Maintenance/Shared/Odometer/Views/OdometerView.swift

```diff
@@ -109,6 +113,52 @@ struct OdometerView: View {
 }
 
 #Preview {
-    OdometerView(userUID: "")
+    var viewModel = OdometerViewModel(userUID: nil)
+    let firstCar = createVehicle(id: "id1", name: "My 1st car")
+    viewModel.vehicles.append(firstCar)
+
+    let secondCar = createVehicle(id: "id2", name: "2nd Car")
+    viewModel.vehicles.append(secondCar)
+    
+    let firstReading = createReading(vehicleID: secondCar.id!,
+                                     date: "2024/10/18",
+                                     distance: 20)
+    viewModel.readings.append(firstReading)
+    
+    let secondReading = createReading(vehicleID: firstCar.id!,
+                                     date: "2024/10/15",
+                                     distance: 1000)
+    viewModel.readings.append(secondReading)
+    
+    let thirdReading = createReading(vehicleID: firstCar.id!,
+                                     date: "2024/10/13",
+                                     distance: 10)
+    viewModel.readings.append(thirdReading)
+
+    return OdometerView(viewModel: viewModel)
         .environment(ActionService.shared)
 }
+
+private func createVehicle(id: String, name: String) -> Vehicle {
```

Comment:
moved these two functions inside the macro, because they'd be local to that scope by default, and we don't need `private`